.github/workflows/tests.yml

```diff
@@ -320,3 +320,15 @@ jobs:
       - run: brew test-bot --only-formulae --test-default-formula
 
       - uses: codecov/codecov-action@29386c70ef20e286228c72b668a06fd0e8399192
+
+      - name: Upload test results to BuildPulse for flaky test detection
+        # Only run this step for PRs where where we have access to secrets.
+        # Run this step even when the tests fail. Skip if the workflow is cancelled.
+        if: github.event.pull_request.head.repo.full_name == github.repository && !cancelled()
+        env:
+          BUILDPULSE_ACCESS_KEY_ID: ${{ secrets.BUILDPULSE_ACCESS_KEY_ID }}
+          BUILDPULSE_SECRET_ACCESS_KEY: ${{ secrets.BUILDPULSE_SECRET_ACCESS_KEY }}
+        run: |
+          curl -fsSL https://github.com/buildpulse/test-reporter/releases/latest/download/test-reporter-darwin-amd64 > ./buildpulse-test-reporter
```

Comment:
Is this something we can package in homebrew-core instead?

Reply:
Agreed. I can try to look into a formula, but I'm struggling to imagine what a decent formula test would be...

Reply:
> I can try to look into a formula, but I'm struggling to imagine what a decent formula test would be...

I'm new to homebrew formula tests, but would it make sense to run the downloaded binary with the `--help` flag and ensure that you get a zero exit status and/or verify that the usage text is emitted?

> The repo also won't pass our notability audit, but I think we can let this slide.

Can you point me toward that audit when you have a moment? Depending on how significant the necessary changes, I may be able to get something in place for this.

Reply:
We don't like `--help` tests, hence my difficulty. We usually want something that tries to exercise some functionality. I guess I could mock a `junit` file (if I knew what those were), pass a fake `--account-id` and `--repository-id`, and see if we get the expected error. Would this be a terrible misuse of your API?

Our audit checks the number of watchers/stars/forks. I think it's one of at least 30, 30, or 75 respectively. I've starred your repo to help this along 😄 That said, I think the reason we want to `brew install` this is so that we use a binary that's versioned and checksummed. I'm sure I don't need to emphasise how important this is for code we're running in CI.

Reply:
> We usually want something that tries to exercise some functionality. I guess I could mock a `junit` file (if I knew what those were), pass a fake `--account-id` and `--repository-id`, and see if we get the expected error. Would this be a terrible misuse of your API?

My hunch is that this will probably work. 😅 Is it important that the test attempts to talk to the BuildPulse API? If not, then there are local errors that we could trigger (e.g., specifying an invalid directory as an arg) to verify the expected error message.

> I think the reason we want to brew install this is so that we use a binary that's versioned and checksummed.

Cool. That makes sense. Can we rely on the GitHub releases URL to identify the version (e.g., https://github.com/buildpulse/test-reporter/releases/download/v0.15.0/test-reporter-darwin-amd64)? 

I'm happy to start publishing checksums as release artifacts going forward if that would help.

Reply:
> there are local errors that we could trigger

Yea, I think those will work. I don't imagine the formula tests would run too often (i.e. only about as often as you cut releases, maybe a bit more), but I'd still feel bad about abusing your API for this. (Unless you appreciate your API being tested for how it deals with invalid calls...)

>I'm happy to start publishing checksums as release artifacts going forward if that would help.

We still need a reliable way to verify the checksums, though, and without a Homebrew formula that would mean hardcoding it in the workflow and then occasionally updating it. That's pretty much the amount of work it'll take to maintain the formula, so we may as well make it available to our users while we're doing so.

Reply:
> > there are local errors that we could trigger
> 
> Yea, I think those will work.

Cool. Thanks for talking through this with me, @carlocab.

Probably the simplest error is to try to submit a non-existent directory:

```
$ test-reporter submit not-a-dir
<buildpulse> Current version: BuildPulse Test Reporter 0.15.0 (darwin be57499 go1.16.4)
<buildpulse> Initiating `submit`
<buildpulse> Received args: not-a-dir
<buildpulse> Using working directory: /usr/local/Homebrew

path is not a directory: not-a-dir

See more help with --help
```

Would that meet the need?

Reply:
Yes. Homebrew/homebrew-core#79806

Thanks for talking me through this as well! By the way, I've built this so that version information appears as
```
❯ buildpulse-test-reporter --version
BuildPulse Test Reporter 0.15.0 (darwin Homebrew go1.16.5)
```
since it seemed like cloning your repo just to get the commit hash is a bit excessive. I can change it to actually reflect the proper commit hash, or some other string, if you prefer.